cli/napi/env.rs

```diff
@@ -52,24 +52,52 @@ fn napi_fatal_exception(env: *mut Env, value: napi_value) -> Result {
   );
 }
 
-// TODO: properly implement
 #[napi_sym::napi_sym]
 fn napi_add_env_cleanup_hook(
-  _env: *mut Env,
-  _hook: extern "C" fn(*const c_void),
-  _data: *const c_void,
+  env: *mut Env,
+  hook: extern "C" fn(*const c_void),
+  data: *const c_void,
 ) -> Result {
-  log::info!("napi_add_env_cleanup_hook is currently not supported");
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+
+  {
+    let mut env_cleanup_hooks = env.cleanup_hooks.borrow_mut();
+    if env_cleanup_hooks
+      .iter()
+      .any(|pair| pair.0 == hook && pair.1 == data)
+    {
+      panic!("Cleanup hook with this data already registered");
+    }
+    env_cleanup_hooks.push((hook, data));
+  }
   Ok(())
 }
 
 #[napi_sym::napi_sym]
 fn napi_remove_env_cleanup_hook(
-  _env: *mut Env,
-  _hook: extern "C" fn(*const c_void),
-  _data: *const c_void,
+  env: *mut Env,
+  hook: extern "C" fn(*const c_void),
+  data: *const c_void,
 ) -> Result {
-  log::info!("napi_remove_env_cleanup_hook is currently not supported");
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+
+  {
+    let mut env_cleanup_hooks = env.cleanup_hooks.borrow_mut();
+    // Hooks are supposed to be removed in LIFO order
+    let maybe_index = env_cleanup_hooks
+      .iter()
+      .rev()
+      .position(|&pair| pair.0 == hook && pair.1 == data);
```

Comment:
Does using `rposition` work? Then this would have the actual index https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.rposition